### PR TITLE
[API] Update API to support querying by start and submission

### PR DESCRIPTION
**Pull request by bfcrampton**
Part of issue https://github.com/DataBiosphere/job-manager/issues/54. Currently we're treating `query.start` as a submission time when querying the pipelines API. `QueryJobsRequest` should support both `start` and `submission` and filter by them accordingly.

This PR just updates the API and regenerates models.

### CODE REVIEW

File: api/jobs.yaml [hunk 1]
```diff
@@ -247,20 +247,24 @@ definitions:
...
+          end datetime.
       start:
         type: string
         format: date-time
         description: >
-          Returns only jobs with an equal or later start datetime. If both start
-          and end date are specified, start date must be before or equal to end
-          date.
+          Returns only jobs with an equal or later start datetime. The start
+          parameter must be prior or equal to the end datetime and end datetime.
```
[the remark this answers is not in the document]
Comment (bfcrampton) [reply api/jobs.yaml]: Whoops, rm.

File: servers/cromwell/jobs/models/query_jobs_result.py [hunk 2]
```diff
@@ -246,7 +246,7 @@ def labels(self, labels):
     def parent_job_id(self):
         """
         Gets the parent_job_id of this QueryJobsResult.
-        Id of the parent job. Expresses a child:parent relationship between this job and the job that called it.
+        Id of the parent job.. Expresses a child:parent relationship between this job and the job that called it.
```
[the remark this answers is not in the document]
Comment (bfcrampton) [reply servers/cromwell/jobs/models/query_jobs_result.py]: I think it was accidentally added in https://github.com/DataBiosphere/job-manager/pull/177/files after regenerating the models. Fixed.

**Comment by bfcrampton** [comment]
> I definitely feel it is odd that of start and submission, we would only support querying by start. However, start is a pretty close approximation so I don't feel having both is much more valuable than one.

That seems like a shame since pipelines doesn't support querying by `start`, so it will likely be less efficient doing all filtering client-side. I would be more inclined to switch to only support querying by `submission` and `end` but I guess that doesn't align with cromwell. 

Is there definitely no value in being able to filter by both `start` vs. `submission`? The only case I can think about is if a user starts 1000s of tasks and doesn't have enough quota to run them all at once. If the user queries by jobs with a start date in the last two weeks, only the jobs which have started will be returned where they may actually want to see all the jobs which were submitted within the last two weeks (which may not be possible using only `start` and `end`).

**Comment by calbach** [comment]
There's definitely some value. `submission` is more useful as not all jobs have a start time, and logically I'm more likely to know what my submission time was and query by it vs "start" which is an arbitrary time at which the system decided to run it.

The downside is that Cromwell may not be able to support it, and it's also slightly more complicated of an API, e.g. the interaction if you specify both submission and start time.

I'd say I'm probably neutral on this, all things considered.

**Comment by calbach** [comment]
And I'll definitely approve, but would like to hear from Broad side on this.

**Comment by bfcrampton** [comment]
Sounds good @samanehsan what do you think about adding `submission`?

**Comment by samanehsan** [comment]
I agree being able to query by `submission` time would be useful, since in Cromwell the workflows are not started immediately. But the Cromwell API currently only allows querying for jobs by start and end time. @geoffjentry is that something the query endpoint could support?

While not as straightforward, it's also possible to filter by status to see which jobs are submitted but not started, or you could add labels to a group of tasks and use that as a filter to track the progress.

**Comment by bfcrampton** [comment]
Closing this PR given that we are making a broader change to our API to handle this need.

**Pull request closed without merging**